owncloudData/src/test/java/com/owncloud/android/data/files/repository/OCFileRepositoryTest.kt

```diff
@@ -25,382 +26,2010 @@ import com.owncloud.android.data.files.datasources.LocalFileDataSource
+            localFileDataSource.saveFile(OC_FILE_WITH_SPACE_ID)
+        }
+    }
+
+    @Test
+    fun `saveConflict saves the etagInConflict related to a file correctly`() {
+        ocFileRepository.saveConflict(OC_FILE_WITH_SPACE_ID.id!!, etagInConflict)
+
+        verify(exactly = 1) {
+            localFileDataSource.saveConflict(
+                fileId = OC_FILE_WITH_SPACE_ID.id!!,
+                eTagInConflict = etagInConflict
+            )
+        }
+    }
+
+    @Test
+    fun `cleanConflict removes conflict for a file correctly`() {
+        ocFileRepository.cleanConflict(OC_FILE_WITH_SPACE_ID.id!!)
+
+        verify(exactly = 1) {
+            localFileDataSource.cleanConflict(OC_FILE_WITH_SPACE_ID.id!!)
+        }
+    }
+
+    @Test
+    fun `disableThumbnailsForFile disables thumbnails for a file correctly`() {
+        ocFileRepository.disableThumbnailsForFile(OC_FILE_WITH_SPACE_ID.id!!)
+
+        verify(exactly = 1) {
+            localFileDataSource.disableThumbnailsForFile(OC_FILE_WITH_SPACE_ID.id!!)
+        }
+    }
+
+    @Test
+    fun `updateFileWithNewAvailableOfflineStatus updates available offline status for a file correctly`() {
+        ocFileRepository.updateFileWithNewAvailableOfflineStatus(OC_FILE_WITH_SPACE_ID, AvailableOfflineStatus.AVAILABLE_OFFLINE)
+
+        verify(exactly = 1) {
+            localFileDataSource.updateAvailableOfflineStatusForFile(
+                ocFile = OC_FILE_WITH_SPACE_ID,
+                newAvailableOfflineStatus = AvailableOfflineStatus.AVAILABLE_OFFLINE
+            )
+        }
+    }
+
+    @Test
+    fun `updateFileWithLastUsage updates last usage for a file correctly`() {
+        val lastUsage = 12345L
+
+        ocFileRepository.updateFileWithLastUsage(OC_FILE_WITH_SPACE_ID.id!!, lastUsage)
+
+        verify(exactly = 1) {
+            localFileDataSource.updateFileWithLastUsage(
+                fileId = OC_FILE_WITH_SPACE_ID.id!!,
+                lastUsage = lastUsage
+            )
+        }
+    }
+
+    @Test
+    fun `updateDownloadedFilesStorageDirectoryInStoragePath updates storage path for downloaded files correctly`() {
+        val oldDirectory = "/old/directory"
+
+        ocFileRepository.updateDownloadedFilesStorageDirectoryInStoragePath(oldDirectory, storagePath)
+
+        verify(exactly = 1) {
+            localFileDataSource.updateDownloadedFilesStorageDirectoryInStoragePath(
+                oldDirectory = oldDirectory,
+                newDirectory = storagePath
+            )
+        }
+    }
+
+    @Test
+    fun `saveDownloadWorkerUuid saves the worker UUID for a file correctly`() {
+        val workerUuid = UUID.randomUUID()
+
+        ocFileRepository.saveDownloadWorkerUuid(OC_FILE_WITH_SPACE_ID.id!!, workerUuid)
+
+        verify(exactly = 1) {
+            localFileDataSource.saveDownloadWorkerUuid(
+                fileId = OC_FILE_WITH_SPACE_ID.id!!,
+                workerUuid = workerUuid
+            )
+        }
+    }
```

Comment:
Maybe it is good to include test cases about network problems, for example (that I have not seen, or I am wrong)

Reply:
We were discussing about this, and we reached the conclusion that we're only going to do tests for non-valid cases when we explicitly catch an exception in the method being tested. That is the case with network errors: we can assume they are caught in upper or lower layers, but this method of this class has nothing to do with it (it will just send it up in case it receives an exception). Moreover, we don't consider it a good practice because we would have to mock an object (for example the datasource) to throw an exception, and the test will check just that, that a mocked object is throwing an exception, nothing to do with the real implementation of the method

Reply:
👍